docs/api.rst

```diff
@@ -812,18 +812,41 @@ to handle it, which defaults to print a traceback and ignoring the exception.
 
 .. function:: on_thread_join(thread)
 
-    Called whenever a thread is joined or created. Note that from the API's perspective there is no way to
-    differentiate between a thread being created or the bot joining a thread.
+    Called whenever the bot joins a thread or gets access to a thread
+    (for example, by gaining access to the parent channel).
 
     Note that you can get the guild from :attr:`Thread.guild`.
 
     This requires :attr:`Intents.guilds` to be enabled.
 
     .. versionadded:: 2.0
 
+    .. versionchanged:: 2.5
+        This is no longer being called when a thread is created, unless the bot itself created
+        the thread (since the thread creator automatically becomes a thread member).
+        See :func:`on_thread_create` instead.
+
     :param thread: The thread that got joined.
     :type thread: :class:`Thread`
 
+.. function:: on_thread_create(thread)
```

Comment:
should be above the docs for on_thread_delete imo

Reply:
ce811dce5731bd89e143bd0dfb45b6e232ccf6c9